src/apply_rules.rs

```diff
@@ -73,6 +76,100 @@ struct PropertyRule {
     property_value: String,
 }
 
+#[derive(Clone, Default, Debug, Deserialize)]
+struct NetworkConsolidation {
+    #[serde(flatten)]
+    network: Network,
+    #[serde(default)]
+    grouped_from: Vec<String>,
+}
+
+fn read_networks_consolidation_file<P: AsRef<Path>>(
+    networks_consolidation_file: P,
+) -> Result<Vec<NetworkConsolidation>> {
+    info!("Reading networks consolidation rules.");
+
+    #[derive(Debug, Deserialize)]
+    struct Consolidation {
+        #[serde(rename = "networks")]
+        networks_consolidation: Vec<NetworkConsolidation>,
+    }
+
+    let file = File::open(networks_consolidation_file)?;
+    let reader = BufReader::new(file);
+    let consolidation: Consolidation =
+        serde_json::from_reader(reader).expect("unvalid networks configuration file");
+    Ok(consolidation.networks_consolidation)
+}
+
+fn check_networks_consolidation(
+    report: &mut Report,
+    networks: &CollectionWithId<Network>,
+    networks_consolidation: Vec<NetworkConsolidation>,
+) -> Result<Vec<NetworkConsolidation>> {
+    info!("Checking networks consolidation.");
+    let mut res: Vec<NetworkConsolidation> = vec![];
+
+    for ntw in networks_consolidation.into_iter() {
+        let mut network_consolidation = false;
+        if networks.get(&ntw.network.id).is_none() {
+            if ntw.grouped_from.is_empty() {
+                report.add_error(
+                    "The grouped network is empty".to_string(),
+                    ReportType::ObjectNotFound,
+                );
+                continue;
+            }
+            for ntw_grouped in &ntw.grouped_from {
+                if networks.get(&ntw_grouped).is_none() {
+                    report.add_error(
+                        format!("The grouped network {} don't exists", ntw_grouped),
```

Comment:
if it's `grouped network`, it should be ` does not exist
otherwise
`The grouped networks {} don't exist`

I'm not sure if it's singular or plural